rqt_bag_plugins/src/rqt_bag_plugins/plot_view.py

```diff
@@ -432,18 +449,31 @@ def _add_msg_object(self, parent, path, name, obj, obj_type):
         if hasattr(obj, '_fields_and_field_types'):
             field_keys = obj.get_fields_and_field_types().keys()
             subobjs = [(field_name, getattr(obj, field_name)) for field_name in field_keys]
-        elif type(obj) in [list, tuple]:
-            len_obj = len(obj)
+            if type(obj) in (Time, TimeMsg):
+                time_obj = obj if type(obj) is Time else Time().from_msg(obj)
+                subobjs.append(('*float_seconds', time_obj.nanoseconds * 1e-9))
+        elif type(obj) in (list, tuple, array.array, numpy.ndarray):
+            if type(obj) in (array.array, numpy.ndarray):
+                list_obj = obj.tolist()
+            else:
+                list_obj = obj
+            len_obj = len(list_obj)
+            short_list_obj = list_obj[:MAX_LIST_LEN]
+
             if len_obj == 0:
                 subobjs = []
             else:
                 w = int(math.ceil(math.log10(len_obj)))
-                subobjs = [('[%*d]' % (w, i), subobj) for (i, subobj) in enumerate(obj)]
+                subobjs = [('[%*d]' % (w, i), subobj) for (i, subobj) in enumerate(short_list_obj)]
+                if len_obj > MAX_LIST_LEN:
+                    for i in range(-LIST_TAIL_LEN, 0):
+                        if len_obj + i >= MAX_LIST_LEN:
+                            subobjs.append(('[%*d]' % (w, len_obj + i), list_obj[i]))
```

Comment:
Can't you use the same logic here as before:
```suggestion
                tail_start = max(MAX_LIST_LEN, len_obj - LIST_TAIL_LEN)
                subobjs.extend([('[%*d]' % (w, i), subobj) for (i, subobj) in enumerate(list_obj[tail_start:])])
```